crates/core/tests/integration.rs

```diff
@@ -0,0 +1,228 @@
+    Error,
+};
+use jsonrpc_core_client::transports::http;
+use solana_sdk::{
+    hash::Hash,
+    message::{Message, VersionedMessage},
+    native_token::LAMPORTS_PER_SOL,
+    pubkey::Pubkey,
+    signature::Keypair,
+    signer::Signer,
+    system_instruction,
+    transaction::VersionedTransaction,
+};
+use std::{str::FromStr, time::Duration};
+use surfpool_core::{
+    rpc::{full::FullClient, minimal::MinimalClient},
+    simnet::{start, SimnetEvent},
+    types::{RpcConfig, RunloopTriggerMode, SimnetConfig, SurfpoolConfig},
+};
+use tokio::time::sleep;
+
+#[tokio::test]
+async fn test_simnet_ready() {
+    let config = SurfpoolConfig {
+        simnet: SimnetConfig {
+            runloop_trigger_mode: RunloopTriggerMode::Manual, // Prevent ticks
+            ..SimnetConfig::default()
+        },
+        ..SurfpoolConfig::default()
+    };
+
+    let (simnet_events_tx, simnet_events_rx) = unbounded();
+    let (_simnet_commands_tx, simnet_commands_rx) = unbounded();
+
+    let _handle = hiro_system_kit::thread_named("test").spawn(move || {
+        let future = start(config, simnet_events_tx, simnet_commands_rx);
+        if let Err(e) = hiro_system_kit::nestable_block_on(future) {
+            panic!("{e:?}");
+        }
+    });
+
+    match simnet_events_rx.recv() {
+        Ok(SimnetEvent::Ready) => println!("Simnet is ready"),
+        e => panic!("Expected Ready event: {e:?}"),
+    }
+}
+
+#[tokio::test]
+async fn test_simnet_ticks() {
+    let config = SurfpoolConfig {
+        simnet: SimnetConfig {
+            slot_time: 1,
+            ..SimnetConfig::default()
+        },
+        ..SurfpoolConfig::default()
+    };
+
+    let (simnet_events_tx, simnet_events_rx) = unbounded();
+    let (_simnet_commands_tx, simnet_commands_rx) = unbounded();
+    let (test_tx, test_rx) = unbounded();
+
+    let _handle = hiro_system_kit::thread_named("test").spawn(move || {
+        let future = start(config, simnet_events_tx, simnet_commands_rx);
+        if let Err(e) = hiro_system_kit::nestable_block_on(future) {
+            panic!("{e:?}");
+        }
+    });
+
+    let _ = hiro_system_kit::thread_named("ticks").spawn(move || {
+        let mut ticks = 0;
+        loop {
+            match simnet_events_rx.recv() {
+                Ok(SimnetEvent::ClockUpdate(_)) => ticks += 1,
+                _ => (),
+            }
+
+            if ticks > 100 {
+                let _ = test_tx.send(true);
+            }
+        }
+    });
+
+    match test_rx.recv_timeout(Duration::from_secs(10)) {
+        Ok(_) => (),
+        Err(_) => panic!("not enough ticks"),
+    }
+}
+
+#[tokio::test]
+async fn test_simnet_some_sol_transfers() {
+    let airdrop_keypairs = (0..10).map(|_| Keypair::new()).collect::<Vec<_>>();
+    let airdrop_addresses: Vec<Pubkey> = airdrop_keypairs.iter().map(|kp| kp.pubkey()).collect();
+    let airdrop_token_amount = LAMPORTS_PER_SOL;
+    let bind_host = "127.0.0.1";
+    let bind_port = 8899;
```

Comment:
This is good for now, but for the future, we may want to find a free port. If we have a full test suite, tests will take forever without parallelization, so we'll need variable ports. I did this in a past project: https://github.com/hirosystems/chainhook/blob/4e4960897983206c376bb7d94a9ba727802ec043/components/chainhook-cli/src/service/tests/helpers/mod.rs#L25-L34